src/OrbitGl/include/OrbitGl/CallTreeView.h

```diff
@@ -192,16 +182,23 @@ class CallTreeView : public CallTreeNode {
  public:
   [[nodiscard]] static std::unique_ptr<CallTreeView> CreateTopDownViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
 
   [[nodiscard]] static std::unique_ptr<CallTreeView>
   CreateBottomUpViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
 
-  CallTreeView() : CallTreeNode{nullptr} {}
+  CallTreeView(const orbit_client_data::ModuleManager* module_manager,
+               const orbit_client_data::CaptureData* capture_data)
+      : CallTreeNode{nullptr}, module_manager_{module_manager}, capture_data_{capture_data} {}
+
+ private:
+  friend CallTreeFunction;
+  const orbit_client_data::ModuleManager* module_manager_;
```

Comment:
Note that we decided to add the two pointers here, and not in the "ItemModel" class, as
1. the view gets created inside OrbitGl, so OrbitQt and the item model does not need to to know about this implementation detail
2. the view stays self-contained

Reply:
I actually started off with the approach of passing `ModuleManager` and `CaptureData` along with `CallTreeView` to the `CallTreeViewItemModel`.
That turned out affect a huge amount of the code base, and a lot of places inside `OrbitQt` (in particular `orbitmainwindow` and a like).

I see the issue with the friend class, but this could be worked around by granting access to the capture data and module manager, to everyone, so that the caller of the `Retrieve...` functions needs to pass in the capture data/module manager directly.

>>the view stays self-contained

>What do you mean?

I mean, you still have just one data class to hold and pass around. The `CallTreeView` is self-contained. It stores all the information to query all the data from it and its child nodes.

Maybe @beckerhe has some more input.

Reply:
> That turned out affect a huge amount of the code base

> you still have just one data class to hold and pass around

So this is exactly my point: it broadens the meaning of `CallTreeView`, from just the root of the tree to a container of different things, but it's just for convenience.

In this case, `CallTreeView` should not inherit from `CallTreeNode` and should be moved to another file. Maybe better, a new container class should be added, with a name that reflects its content. This should be evaluated.

At that point, you can pass the two pointers directly to the `Retrieve...` methods, fetched from `CallTreeView` or the new container class, instead of the "whole" `CallTreeView` or the new container class, because it's weird (and not necessary) to pass the whole tree back to a node of the tree.

But again, I'd still consider simply keeping the two pointers separate if it's feasible, as adding yet a new container with its own new semantics is just more complicated to understand.

> I see the issue with the friend class, but this could be worked around by granting access to the capture data and module manager

My point is that `friend` is not the problem, it's the manifestation of the problem mentioned. Only making the pointers public won't really change that. 
In any case, as mentioned above, if the pointers are exposed then you can directly pass the pointers to the `Retrieve...` methods.

Reply:
> CallTreeView should not inherit from CallTreeNode and should be moved to another file

Looking at this class again, this is a candidate. The current inheritance is confusing, `CallTreeView` is conceptually not a `CallTreeNode`.

Reply:
Done. Still kept the stuff in the same cpp file (for now). I can move stuff around if you prefer that, but wanted to give you a sneak peek first.